compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1425,7 +1425,16 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                             DefPathData::ImplTrait,
                             span,
                         );
-                        let ident = Ident::from_str_and_span(&pprust::ty_to_string(t), span);
+
+                        // HACK: pprust breaks strigns with newlines when the type
+                        // gets too long. We don't want these to show up in compiler
+                        // output or built artifacts, so replace them here...
+                        // Perhaps we should instead format APITs more robustly.
```

Comment:
Can we change this in pprust instead? I don't know why we'd want it to add the newline ... if it's just for -Zunpretty=expanded that doesn't seem worth adding hacks for.

Reply:
This newline is not `impl Trait` specifically. This is just the regular line breaking that pprust does for code that is too long:

https://github.com/rust-lang/rust/blob/096309e6dc4378e175e524e963332d1a3830ed63/compiler/rustc_ast_pretty/src/pp.rs#L417-L435

I don't think we should remove pprust's ability to break lines, and I don't think it's worth it to introduce extra configurables (like breaking only with spaces). I'd rather have the hack be as localized as possible, or else rewrite this formatting code to no longer rely on pprust at all.